src/main/kotlin/uk/gov/justice/digital/hmpps/externalusersapi/jpa/repository/UserRepository.kt

```diff
@@ -1,15 +1,33 @@
 package uk.gov.justice.digital.hmpps.externalusersapi.jpa.repository
 
-import org.springframework.data.jpa.repository.JpaSpecificationExecutor
-import org.springframework.data.jpa.repository.Query
-import org.springframework.data.repository.CrudRepository
-import uk.gov.justice.digital.hmpps.externalusersapi.model.User
+import kotlinx.coroutines.flow.Flow
+import org.springframework.data.r2dbc.repository.Query
+import org.springframework.data.repository.kotlin.CoroutineCrudRepository
+import org.springframework.lang.NonNull
+import reactor.core.publisher.Mono
+import uk.gov.justice.digital.hmpps.externalusersapi.r2dbc.data.User
 import uk.gov.justice.digital.hmpps.externalusersapi.security.AuthSource
-import java.util.Optional
 import java.util.UUID
 
-interface UserRepository : CrudRepository<User, UUID>, JpaSpecificationExecutor<User> {
-  @Query("select distinct u from User u left join fetch u.authorities where u.username = :username and u.source = :source")
-  fun findByUsernameAndSource(username: String?, source: AuthSource): Optional<User>
-  fun findByUsername(username: String?): Optional<User> = findByUsernameAndSource(username, AuthSource.auth)
+interface UserRepository : CoroutineCrudRepository<User, UUID> {
+
+  @NonNull
+  @Query(
+    """ select * from  USERS u
+            where
+              u.username = :username
+                  and
+              u.source = :source """
+  )
+  suspend fun findByUsernameAndSource(username: String?, source: AuthSource = AuthSource.auth): Mono<User>
```

Comment:
Are we currently allowing calls to external users for username and AuthSource.nomis?  if not, we can reduce this to just no include the authsource